apps/explorer/src/pages/AddressInfoPage/AssetList.tsx

```diff
@@ -50,30 +51,51 @@ const AssetList = ({ addressHash, addressBalance, limit, className }: AssetListP
   const {
     fungibleTokens: fungibleTokensMetadata,
     nfts: nftsMetadata,
-    unknown: unknownAssetsIds,
-    isLoading: assetsMetadataLoading
+    unknown: unknownAssetsIds
   } = useAssetsMetadata(assetIds)
 
-  const isLoading = assetsLoading || assetsMetadataLoading
+  const tokensWithAvailablePrice = useTokensWithAvailablePrice()
+  const tokensPrices = useTokensPrices([
+    ALPH.symbol,
+    ...fungibleTokensMetadata.flatMap((t) => (tokensWithAvailablePrice.includes(t.symbol) && t.symbol) || [])
+  ])
+
+  const isLoading = assetsLoading
 
   const fungibleTokens = useMemo(() => {
     const unsorted = flatMap(fungibleTokensMetadata, (t) => {
       const balance = assets?.find((a) => a.tokenId === t.id)
 
-      return balance ? [{ ...t, balance: BigInt(balance.balance), lockedBalance: BigInt(balance.lockedBalance) }] : []
+      return balance
+        ? [
+            {
+              ...t,
+              balance: BigInt(balance.balance),
+              lockedBalance: BigInt(balance.lockedBalance),
+              worth: (t.verified && calculateAmountWorth(BigInt(balance.balance), tokensPrices[t.symbol])) || undefined
+            }
+          ]
+        : []
     })
 
     // Add ALPH
     if (addressBalance && BigInt(addressBalance.balance) > 0) {
       unsorted.unshift({
         ...alphMetadata,
         balance: BigInt(addressBalance.balance),
-        lockedBalance: BigInt(addressBalance.lockedBalance)
+        lockedBalance: BigInt(addressBalance.lockedBalance),
+        worth: calculateAmountWorth(BigInt(addressBalance.balance), tokensPrices[ALPH.symbol]) || undefined
       })
     }
 
-    return sortBy(unsorted, [(t) => !t.verified, (t) => !t.name, (t) => t.name.toLowerCase(), 'id'])
-  }, [addressBalance, assets, fungibleTokensMetadata])
+    return sortBy(unsorted, [
+      (t) => !t.verified,
+      (t) => t.worth && 1 / t.worth,
+      (t) => !t.name,
+      (t) => t.name.toLowerCase(),
+      'id'
+    ])
```

Comment:
We have a function called `sortAssets` in the shared package that does this.

https://github.com/alephium/alephium-frontend/blob/master/packages/shared/src/assets.ts#L24-L35

Reply:
Done